app/controllers/messages_controller.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class MessagesController < ApplicationController
+  before_action :authenticate_user!
+
+  def create
+    message = conversation.messages.new(message_params.merge(user_id: current_user.id))
+
+    if message.save
+      render json: message, status: :created
+    else
+      render json: { errors: message.errors.messages }, status: :unprocessable_entity
+    end
+  end
+
+  private
+
+  def conversation
```

Comment:
If you eventually implement json builder to use the views, consider declaring this as a helper method so you can access to the conversation in the view file